### PR TITLE
fix(core): source install

### PR DESCRIPTION
**Pull request by sc979**
## Description

Check that gorgone is already installed (mandatory)
Add gorgone user to each centreon's group
Add centreon to gorgone group
In case of upgrade : 
    - Copy the ssh keys to a specific gorgone folder

**Fixes** # (none)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [ ] 19.10.x
- [x] 20.04.x (master)

<h2> How this pull request can be tested ? </h2>

Upgrade to the 20.04.x using the source script

#### Community contributors & Centreon team

- [x] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: libinstall/CentWeb.sh [hunk 1]
```diff
@@ -119,6 +118,13 @@ add_group "$WEB_USER" "$MONITORINGENGINE_GROUP"
 add_group "$CENTREON_USER" "$MONITORINGENGINE_GROUP"
 add_group "$CENTREON_USER" "$WEB_GROUP"
 
+## Configure Gorgone user and group
+add_group "$CENTREON_USER" "$GORGONE_GROUP"
```
Comment (sc979) [inline libinstall/CentWeb.sh]:
```suggestion
add_group "$CENTREON_USER" "$GORGONE_GROUP"
add_group "$WEB_USER" "$GORGONE_GROUP"
```

File: install.sh [hunk 2]
```diff
@@ -263,12 +281,14 @@ if [ "$upgrade" -eq 1 ] ; then
...
 fi
 
-if [ "$silent_install" -ne 1 ] ; then 
+if [ "$silent_install" -ne 1 ] ; then
 	echo "$line"
 	echo -e "\t$(gettext "Please choose what you want to install")"
 	echo "$line"
 fi
 
+export upgrade
```
Comment (kduret) [inline install.sh]: $upgrade is not already exported ?

File: libinstall/functions [hunk 3]
```diff
@@ -1218,6 +1232,45 @@ configureApache() {
...
 }
 
+#----
+## Copy ssh keys to gorgone and add proper rights
+#----
+copy_ssh_keys_to_gorgone() {
+    if [ ! -d $GORGONE_VARLIB ] ; then
+        mkdir -p $GORGONE_VARLIB/.ssh
+        if [ "$?" -nq 0 ] ; then
+            echo_failure "$(gettext "Cannot create the surfolder $GORGONE_VARLIB/.ssh")" "$fail"
```
Comment (kduret) [inline libinstall/functions]:
```suggestion
            echo_failure "$(gettext "Cannot create the subfolder $GORGONE_VARLIB/.ssh")" "$fail"
```

File: install.sh (hunk 2 above)
Comment (sc979) [reply install.sh]: nope, "inst_upgrade_dir" and other variables are exported here :
https://github.com/centreon/centreon/pull/8396/files/aa6a8ba9250af654f7cd1f71316eb2848e354226#diff-3fbb47e318cd8802bd325e7da9aaabe8R168
but these aren't directly relevant with the upgrade process

**Pull request merged**